mdshack/shotshare
# Update env variables to get docker compose to run on initial build

### Description
Tried Local Development instructions on the README but ran into some issues. Adding/changing these environment variables solved the issue for me.

[Added](https://github.com/mdshack/shotshare/pull/47/files#diff-a3046da0d15a27e89f2afe639b25748a7ad4d9290af3e7b1b6c1a5533c8f0a8cR6) `APP_SERVICE` to match the [change in the docker-compose file](https://github.com/laravel/sail/issues/263#issuecomment-1597817348)

[Changed to docker endpoint to be able to connect via a Database viewer](https://github.com/mdshack/shotshare/pull/47/files#diff-a3046da0d15a27e89f2afe639b25748a7ad4d9290af3e7b1b6c1a5533c8f0a8cR13)

[Added ](https://github.com/mdshack/shotshare/pull/47/files#diff-a3046da0d15a27e89f2afe639b25748a7ad4d9290af3e7b1b6c1a5533c8f0a8cR62-R63)`WWWGROUP` & `WWWUSER` ids because running `docker compose up` on the initial image build [tries to map sail specific attributes on the build](https://laracasts.com/discuss/channels/laravel/groupadd-invalid-group-id-sail). This isn't an issue if you run `./vendor/bin/sail up` on the initial build because `docker compose` runs fine after

### Testing Instructions
1. Run `docker compose up --build` without `WWWGROUP` & `WWWUSER` and it'll error out.
2. Run the same command after adding the env variables and it should work.
